python/common/org/python/types/Str.java

```diff
@@ -384,6 +384,27 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     @org.python.Method(
             __doc__ = ""
     )
+    public org.python.Object __floordiv__(org.python.Object other) {
+        if (other instanceof org.python.types.Complex) {
```

Comment:
I believe this is not a behavior specific to strings, see:

```
>>> 2 // complex(1, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't take floor of complex number.
>>>
>>> object() // complex(1, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't take floor of complex number.
```

Have you tried using this code in the `Object` parent class instead?